src/test/java/com/hotels/beeju/extensions/HiveServer2JUnitExtensionTest.java

```diff
@@ -0,0 +1,276 @@
+  }
+
+  @Test
+  public void dropDatabase() throws Exception {
+    String databaseName = "Another_DB";
+
+    server.createDatabase(databaseName);
+    try (Connection connection = DriverManager.getConnection(server.connectionURL());
+        Statement statement = connection.createStatement()) {
+      String dropHql = String.format("DROP DATABASE %s", databaseName);
+      statement.execute(dropHql);
+    }
+
+    HiveMetaStoreClient client = server.newClient();
+    try {
+      client.getDatabase(databaseName);
+      fail(String.format("Database %s was not deleted", databaseName));
+    } catch (NoSuchObjectException e) {
+      // expected
+    } finally {
+      client.close();
+    }
+  }
+
+  @Test
+  public void createTable() throws Exception {
+    String tableName = "my_test_table";
+
+    try (Connection connection = DriverManager.getConnection(server.connectionURL());
+        Statement statement = connection.createStatement()) {
+      String createHql = new StringBuilder()
+          .append("CREATE TABLE `" + DATABASE + "." + tableName + "`(`id` int, `name` string) ")
+          .append("ROW FORMAT SERDE 'org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe' ")
+          .append("STORED AS INPUTFORMAT 'org.apache.hadoop.mapred.TextInputFormat' ")
+          .append("OUTPUTFORMAT 'org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat'")
+          .toString();
+      statement.execute(createHql);
+    }
+
+    HiveMetaStoreClient client = server.newClient();
+    Table table = client.getTable(DATABASE, tableName);
+    client.close();
+    assertThat(table.getDbName(), is(DATABASE));
+    assertThat(table.getTableName(), is(tableName));
+    assertThat(table.getSd().getCols(),
+        is(Arrays.asList(new FieldSchema("id", "int", null), new FieldSchema("name", "string", null))));
+    assertThat(table.getSd().getInputFormat(), is("org.apache.hadoop.mapred.TextInputFormat"));
+    assertThat(table.getSd().getOutputFormat(), is("org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat"));
+    assertThat(table.getSd().getSerdeInfo().getSerializationLib(),
+        is("org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe"));
+  }
+
+  @Test
+  public void dropTable() throws Exception {
+    String tableName = "my_table";
+    createUnpartitionedTable(DATABASE, tableName);
+
+    try (Connection connection = DriverManager.getConnection(server.connectionURL());
+        Statement statement = connection.createStatement()) {
+      String dropHql = String.format("DROP TABLE %s.%s", DATABASE, tableName);
+      statement.execute(dropHql);
+    }
+
+    HiveMetaStoreClient client = server.newClient();
+    try {
+      client.getTable(DATABASE, tableName);
+      fail(String.format("Table %s.%s was not deleted", DATABASE, tableName));
+    } catch (NoSuchObjectException e) {
+      // expected
+    } finally {
+      client.close();
+    }
+  }
+
+  @Test
+  public void addPartition() throws Exception {
```

Comment:
There is lots of duplication in between this class and the `com.hotels.beeju.HiveServer2JUnitRuleTest`. Perhaps some of those tests can move to `com.hotels.beeju.core.HiveServer2CoreTest` and the Rule/Extension test can test just the logic of the Rule/Extension?

Reply:
Yeah I had a similar thought when looking at it. But I think the issue with doing that is these tests all use different elements of the rule/extension which isn't available in the `HiveServer2Core`, such as the temporary directory and the server connection URL. Maybe some of the reused logic could be moved into a test resources class?

Reply:
I think it's better to move it to a core test as it is core logic we're testing. The core test should just provide a temporary directory and a server connection url.